core/arch/arm/plat-ti/main.c

```diff
@@ -130,12 +93,16 @@ struct plat_nsec_ctx {
 
 void init_sec_mon(unsigned long nsec_entry)
 {
-	struct plat_nsec_ctx *plat_ctx = (struct plat_nsec_ctx *)nsec_entry;
+	struct plat_nsec_ctx *plat_ctx;
 	struct sm_nsec_ctx *nsec_ctx;
 
+	plat_ctx = phys_to_virt(nsec_entry, MEM_AREA_IO_SEC);
```

Comment:
@glneo: using MEM_AREA_IO_SEC or MEM_AREA_RAM_SEC is just a... trick to define physical address ranges as cached or uncached (both secure of course), whatever the physical HW lies below. So it's fine using MEM_AREA_IO_SEC for uncached RAM.

- If your expected data were written through the cache, then you should map it cached and no need to flush or invalidate it previously.
- If the data were written uncached, then map it uncached and no need to flush or invalidate.
- If your not unsure... it's a tricky situation. Mapping cached and invalidating cache may discard content that were cached. 

I think you should have a look how the earlier boot stage handled these data. Or maybe you should flush the overall cache at optee entrypoint. Actually the earlier boot stage should have flushed its cache for data accessed by its next boot stage before entering it (here optee).